src/overlays/actors/ovl_En_Ge1/z_en_ge1.c

```diff
@@ -28,45 +27,431 @@ const ActorInit En_Ge1_InitVars = {
+#define GERUDO_WHITE_STATE_DISABLE_MOVEMENT 8
 
-extern UNK_TYPE D_06002B98;
+typedef enum {
+    /* 0 */ GERUDO_WHITE_HAIR_BOB,
+    /* 1 */ GERUDO_WHITE_HAIR_STRAIGHT,
+    /* 2 */ GERUDO_WHITE_HAIR_SPIKEY
+} GerudoWhiteHairstyle;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge1/EnGe1_Init.s")
+typedef enum {
+    /* -1 */ GERUDO_WHITE_ANIM_NONE = -1,
+    /*  0 */ GERUDO_WHITE_ANIM_ARMS_FOLDED,
+    /*  1 */ GERUDO_WHITE_ANIM_UNFOLDING_ARMS, // Unused
+    /*  2 */ GERUDO_WHITE_ANIM_STANDING_HEAD_BOWED,
+    /*  3 */ GERUDO_WHITE_ANIM_STIFFENING,
+    /*  4 */ GERUDO_WHITE_ANIM_STIFF_SHIVERING,
+    /*  5 */ GERUDO_WHITE_ANIM_TRUDGING_OFF,
+    /*  6 */ GERUDO_WHITE_ANIM_EXCITED_CLAPPING,
+    /*  7 */ GERUDO_WHITE_ANIM_SALUTE,
+    /*  8 */ GERUDO_WHITE_ANIM_LEADING_BOAT,
+    /*  9 */ GERUDO_WHITE_ANIM_BLOWN_AWAY,
+} GerudoWhiteAnimations;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge1/EnGe1_Destroy.s")
+void EnGe1_SetAnimation(EnGe1* this, s16 index, u8 mode, f32 morphFrames);
+void EnGe1_ShadowDraw(Actor* actor, Lights* lights, PlayState* play);
+void EnGe1_Wait(EnGe1* this, PlayState* play);
+void EnGe1_PerformCutsceneActions(EnGe1* this, PlayState* play);
+s32 EnGe1_ValidatePictograph(PlayState* play, Actor* this);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge1/func_809457EC.s")
+void EnGe1_Init(Actor* thisx, PlayState* play) {
+    EnGe1* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge1/func_80945924.s")
+    ActorShape_Init(&this->picto.actor.shape, 0.0f, EnGe1_ShadowDraw, 30.0f);
+    SkelAnime_InitFlex(play, &this->skelAnime, &gGerudoWhiteSkel, &gGerudoWhiteArmsFoldedAnim, this->jointTable,
+                       this->morphTable, 16);
+    Collider_InitAndSetCylinder(play, &this->collider, &this->picto.actor, &sCylinderInit);
+    this->picto.actor.colChkInfo.mass = MASS_IMMOVABLE;
+    this->picto.actor.targetMode = 6;
+    Actor_SetScale(&this->picto.actor, 0.01f);
+    this->curAnim = this->csAction = -1; // GERUDO_WHITE_ANIM_NONE
+    this->stateFlags = 0;
+    EnGe1_SetAnimation(this, GERUDO_WHITE_ANIM_ARMS_FOLDED, ANIMMODE_LOOP, 0.0f);
+    this->actionFunc = EnGe1_Wait;
+    this->picto.validationFunc = EnGe1_ValidatePictograph;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge1/func_80945A00.s")
+    switch (GERUDO_WHITE_GET_TYPE(&this->picto.actor)) {
+        default:
+            this->hairstyle = GERUDO_WHITE_HAIR_BOB;
+            break;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge1/func_80945A9C.s")
+        case 1:
+            this->hairstyle = GERUDO_WHITE_HAIR_STRAIGHT;
+            break;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge1/func_80945B60.s")
+        case 2:
+            this->hairstyle = GERUDO_WHITE_HAIR_SPIKEY;
+            break;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ge1/func_80945C50.s")
+        case 0: // Only type used in-game
```

Comment:
Could probably make an enum for the type as well.

Reply:
I don't think there's much point here, it's used once, only one is used, and it doesn't really do anything (the OoT ones have lots of different ones that set the same hairstyle: https://github.com/zeldaret/oot/blob/9647f50451f26fa8b96653bf792f86c6148f6d12/src/overlays/actors/ovl_En_Ge1/z_en_ge1.c#L107-L173 )

Reply:
I think it is still valuable. It would be the outward facing enum that someone could use to spawn each type.

Reply:
And call them what? USED, UNUSED_1, UNUSED_2?

Reply:
Personally I would just name them after their hairstyle, since that seems to distinguish them all.